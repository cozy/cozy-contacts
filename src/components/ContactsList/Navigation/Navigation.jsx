import { ButtonLink } from 'cozy-ui/transpiled/react/Button'
import React, { useEffect, useState } from 'react'
import PropTypes from 'prop-types'

const ANCHOR_REGEX = /#(.)$/

const Navigation = React.memo(({ letters }) => {
  Navigation.displayName = 'Navigation'
  const [selected, setSelected] = useState(null)

  useEffect(
    () => {
      if (letters.length > 0) {
        const matches = window.location.href.match(ANCHOR_REGEX)
        const defaultValue =
          (matches && matches.length === 2 && matches[1]) || null

        if (defaultValue) {
          window.location.href = `/#${defaultValue}`
          setSelected(defaultValue)
        }
      }
    },
    [letters]
  )

  return (
    <nav className="navigation">
      <ol>
        {letters.map(letter => {
          return (
            <ButtonLink
              key={letter}
              onClick={() => setSelected(letter)}
              label={letter}
              theme={letter === selected ? 'regular' : 'secondary'}
              href={`#${letter}`}
              round
            />
          )
        })}
      </ol>
    </nav>
  )
})

Navigation.propTypes = {
  letters: PropTypes.arrayOf(PropTypes.string).isRequired
}
Navigation.defaultProps = {
  letters: []
}

export default Navigation
