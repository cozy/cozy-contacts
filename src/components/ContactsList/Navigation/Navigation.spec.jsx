import React from 'react'
import renderer from 'react-test-renderer'
import Navigation from './Navigation'
import { mount } from 'enzyme'
import { render } from '@testing-library/react'

describe('Navigation component', () => {
  it('render correctly navigation component', () => {
    const NavigationComponent = renderer
      .create(<Navigation letters={[]} />)
      .toJSON()
    expect(NavigationComponent).toMatchSnapshot()
  })

  it('render navigation correctly with default props for letters', () => {
    const NavigationComponent = mount(<Navigation />)
    expect(NavigationComponent.prop('letters')).toEqual([])
  })

  it('render navigation correctly with letters', () => {
    const NavigationComponent = mount(<Navigation letters={['A', 'B']} />)
    expect(NavigationComponent.prop('letters')).toEqual(['A', 'B'])
  })

  it('Navigation component renders button link A and B', () => {
    const { getByText } = render(<Navigation letters={['A', 'B']} />)
    expect(getByText('A'))
    expect(getByText('B'))
  })

  it('Navigation component renders one button link with anchor in href', () => {
    render(<Navigation letters={['A']} />)
    document.querySelectorAll('a')
    expect(document.querySelector('a').getAttribute('href')).toBe('#A')
  })
})
