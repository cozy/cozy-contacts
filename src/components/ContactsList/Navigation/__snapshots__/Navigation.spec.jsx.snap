// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component render correctly navigation component 1`] = `
<nav
  className="navigation"
>
  <ol />
</nav>
`;
